metriq-api/service/propertyService.js

```diff
@@ -0,0 +1,102 @@
+    platformDataType.dataTypeId = property.dataTypeId
+    platformDataType.userId = userId
+
+    const platformDataTypeValidateResponse = await platformDataTypeService.validate(platformDataType)
+    if (!platformDataTypeValidateResponse.success) {
+      return platformDataTypeValidateResponse
+    }
+
+    const platformDataTypeValue = (await platformDataTypeValueService.SequelizeServiceInstance.new())
+    platformDataTypeValue.value = property.value
+    platformDataTypeValue.platformId = platformId
+    platformDataTypeValue.notes = property.valueDescription ? property.valueDescription : ''
+    platformDataTypeValue.userId = userId
+
+    const platformDataTypeValueValidateResponse = await platformDataTypeValueService.validate(platformDataTypeValue)
+    if (!platformDataTypeValueValidateResponse.success) {
+      return platformDataTypeValueValidateResponse
+    }
+
+    const platformDataTypeCreateResponse = await platformDataTypeValueService.create(platformDataType)
+    if (!platformDataTypeCreateResponse.success) {
+      return platformDataTypeCreateResponse
+    }
+
+    platformDataTypeValue.platformDataTypeId = platformDataType.id
+
+    const platformDataTypeValueCreateResponse = await platformDataTypeValueService.create(platformDataTypeValue)
+    if (!platformDataTypeValueCreateResponse.success) {
+      return platformDataTypeValueCreateResponse
+    }
+
+    property.id = platformDataType.id
+
+    return { success: true, body: property }
+  }
+
+  async update (propertyId, property, userId) {
+    const platformDataTypeValue = await platformDataTypeValueService.getByPk(propertyId)
+    if (!platformDataTypeValue) {
+      return { success: false, message: 'Property ID not found.' }
+    }
+    const platformDataType = await platformDataTypeValue.getPlatformDataType()
+
+    platformDataType.name = property.name
+    platformDataType.fullName = property.fullName ? property.fullName : property.name
+    platformDataType.description = property.typeDescription ? property.typeDescription : ''
+    platformDataType.dataTypeId = property.dataTypeId
+    platformDataType.userId = userId
+
+    const platformDataTypeValidateResponse = await platformDataTypeService.validate(platformDataType)
+    if (!platformDataTypeValidateResponse.success) {
+      return platformDataTypeValidateResponse
+    }
+
+    platformDataTypeValue.value = property.value
+    platformDataTypeValue.notes = property.valueDescription ? property.valueDescription : ''
+    platformDataTypeValue.userId = userId
+
+    const platformDataTypeValueValidateResponse = await platformDataTypeValueService.validate(platformDataTypeValue)
+    if (!platformDataTypeValueValidateResponse.success) {
+      return platformDataTypeValueValidateResponse
+    }
+
+    await platformDataTypeValue.save()
+    await platformDataType.save()
+
+    return { success: true, body: property }
+  }
+
+  async getAllNames () {
+    return await platformDataTypeService.getAll()
```

Comment:
Not sure if this is intentional--is this getting the names? It might be but I can't definitively say.

Reply:
It is. Our `/api/[model]/names` route return slightly more information than the "name," like the primary key, but this serves the same function as all `.../names` routes, appropriate to the corresponding model.